src/main/java/competition/subsystems/drive/commands/SwerveToNearestScoringPositionCommand.java

```diff
@@ -1,36 +1,66 @@
 package competition.subsystems.drive.commands;
 
 import competition.auto_programs.AutoLandmarks;
+import competition.operator_interface.OperatorInterface;
 import competition.subsystems.drive.DriveSubsystem;
 import competition.subsystems.pose.PoseSubsystem;
 import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Transform2d;
 import edu.wpi.first.wpilibj.DriverStation;
+import xbot.common.math.WrappedRotation2d;
 import xbot.common.math.XYPair;
+import xbot.common.properties.DoubleProperty;
 import xbot.common.properties.PropertyFactory;
 import xbot.common.subsystems.drive.control_logic.HeadingModule;
+import xbot.common.subsystems.feedback.XRumbleManager;
 
 import javax.inject.Inject;
 import java.util.Arrays;
 import java.util.List;
 
 public class SwerveToNearestScoringPositionCommand extends SwerveToPointCommand {
 
+    private final DoubleProperty maxTravelDistanceProp;
+    private final XRumbleManager rumbleManager;
+
     private Pose2d targetPose;
 
     @Inject
     public SwerveToNearestScoringPositionCommand(
             DriveSubsystem drive,
             PoseSubsystem pose,
             PropertyFactory pf,
-            HeadingModule.HeadingModuleFactory headingModuleFactory
+            HeadingModule.HeadingModuleFactory headingModuleFactory,
+            OperatorInterface oi
     ) {
         super(drive, pose, pf, headingModuleFactory);
+        this.rumbleManager = oi.driverGamepad.getRumbleManager();
+
+        maxTravelDistanceProp = pf.createPersistentProperty("Maximum travel distance inches", 100);
+
+        this.setFieldRelativeMotion();
     }
 
     @Override
     public final void initialize() {
-        this.targetPose = getTargetPose();
-        setTargetPosition(new XYPair(targetPose.getX(), targetPose.getY()), targetPose.getRotation().getDegrees());
+        Pose2d currentPose = this.pose.getCurrentPose2d();
+        Pose2d newPose = getTargetPose();
+        targetPose = newPose;
+        log.info(String.format("Current pose is: %s; Target pose is: %s", currentPose.toString(), this.targetPose.toString()));
+
+        Transform2d travel = newPose.minus(currentPose);
```

Comment:
Not sure which approach is better, but 

`pose.getCurrentPose2d().getTranslation().getDistance(getTargetPose().getTranslation());` is also an option.